src/rwe/CursorService.cpp

```diff
@@ -3,60 +3,52 @@
-        TimeService* timeService,
-        std::shared_ptr<SpriteSeries> normalCursor,
-        std::shared_ptr<SpriteSeries> selectCursor,
-        std::shared_ptr<SpriteSeries> attackCursor,
-        std::shared_ptr<SpriteSeries> moveCursor,
-        std::shared_ptr<SpriteSeries> redCursor)
-        : sdlContext(sdlContext),
-          timeService(timeService),
-          _normalCursor(std::move(normalCursor)),
-          _selectCursor(std::move(selectCursor)),
-          _attackCursor(std::move(attackCursor)),
-          _moveCursor(std::move(moveCursor)),
-          _redCursor(std::move(redCursor)),
-          currentCursor(_normalCursor.get())
+        SdlContext&     sdlContext,
+        TimeService&    timeService,
+        TextureService& textureService)
+        : _sdlContext(sdlContext),
+          _timeService(timeService),
+          _textureService(textureService),
+          _currentCursor(Type::normal)
     {
+        _cursors[Type::move]      = textureService.getGafEntry("anims/CURSORS.GAF", "cursormove");
+        _cursors[Type::grn]       = textureService.getGafEntry("anims/CURSORS.GAF", "cursorgrn");
+        _cursors[Type::select]    = textureService.getGafEntry("anims/CURSORS.GAF", "cursorselect");
+        _cursors[Type::red]       = textureService.getGafEntry("anims/CURSORS.GAF", "cursorred");
+        _cursors[Type::load]      = textureService.getGafEntry("anims/CURSORS.GAF", "cursorload");
+        _cursors[Type::revive]    = textureService.getGafEntry("anims/CURSORS.GAF", "cursorrevive");
+        _cursors[Type::defend]    = textureService.getGafEntry("anims/CURSORS.GAF", "cursordefend");
+        _cursors[Type::patrol]    = textureService.getGafEntry("anims/CURSORS.GAF", "cursorpatrol");
+        _cursors[Type::protect]   = textureService.getGafEntry("anims/CURSORS.GAF", "cursorprotect");
+        _cursors[Type::repair]    = textureService.getGafEntry("anims/CURSORS.GAF", "cursorrepair");
+        _cursors[Type::attack]    = textureService.getGafEntry("anims/CURSORS.GAF", "cursorattack");
+        _cursors[Type::normal]    = textureService.getGafEntry("anims/CURSORS.GAF", "cursornormal");
+        _cursors[Type::pickup]    = textureService.getGafEntry("anims/CURSORS.GAF", "cursorpickup");
+        _cursors[Type::airstrike] = textureService.getGafEntry("anims/CURSORS.GAF", "cursorairstrike");
+        _cursors[Type::teleport]  = textureService.getGafEntry("anims/CURSORS.GAF", "cursorteleport");
+        _cursors[Type::reclamate] = textureService.getGafEntry("anims/CURSORS.GAF", "cursorreclamate");
+        _cursors[Type::findsite]  = textureService.getGafEntry("anims/CURSORS.GAF", "cursorfindsite");
+        _cursors[Type::unload]    = textureService.getGafEntry("anims/CURSORS.GAF", "cursorunload");
+        _cursors[Type::hourglass] = textureService.getGafEntry("anims/CURSORS.GAF", "cursorhourglass");
+        _cursors[Type::toofar]    = textureService.getGafEntry("anims/CURSORS.GAF", "cursortoofar");
```

Comment:
RWE style is to never do work in constructors, particularly IO -- we only map parameters to members. The work of loading the cursors should be done by the caller or by a factory method.